src/codewhisperer/commands/gettingStartedPageCommands.ts

```diff
@@ -6,13 +6,20 @@ import * as vscode from 'vscode'
 import { CommandDeclarations, Commands } from '../../shared/vscode/commands2'
 import { showCodeWhispererWebview, CodeWhispererSource } from '../vue/backend'
 import { telemetry } from '../../shared/telemetry/telemetry'
+
+import { PromptSettings } from '../../shared/settings'
+
 /**
  * The methods with backend logic for the Codewhisperer Getting Started Page commands.
  */
 export class CodeWhispererCommandBackend {
     constructor(private readonly extContext: vscode.ExtensionContext) {}
-    public showGettingStartedPage(source: CodeWhispererSource) {
-        telemetry.ui_click.emit({ elementId: 'codewhisperer_Learn_ButtonClick', passive: true })
+    public async showGettingStartedPage(source: CodeWhispererSource) {
+        const prompts = PromptSettings.instance
+        //To check the condition If the user has already seen the welcome message
+        if (!(await prompts.isPromptEnabled('codeWhispererNewWelcomeMessage'))) {
+            telemetry.ui_click.emit({ elementId: 'codewhisperer_Learn_ButtonClick', passive: true })
+        }
```

Comment:
I am not following, why do we need to check the condition here?

Reply:
We need to check the condition because this function is called in both cases 
When user trying to open the Learn UX using Button click and 
When user is  trying open automatically for the first time 

To differentiate 2 scenarios we need a check for this

Reply:
ah ok.